go-controller/pkg/ovn/namespace.go

```diff
@@ -212,22 +217,24 @@ func parseRoutingExternalGWAnnotation(annotation string) ([]net.IP, error) {
 	return routingExternalGWs, nil
 }
 
-// AddNamespace creates corresponding addressset in ovn db
-func (oc *Controller) AddNamespace(ns *kapi.Namespace) {
+// addNamespace creates corresponding addressset in ovn db
+func (oc *Controller) addNamespace(key string, ns *kapi.Namespace) error {
 	klog.Infof("[%s] adding namespace", ns.Name)
 	// Keep track of how long syncs take.
 	start := time.Now()
 	defer func() {
+		oc.cacheNamespace(key, ns)
 		klog.Infof("[%s] adding namespace took %v", ns.Name, time.Since(start))
 	}()
 
 	_, nsUnlock, err := oc.ensureNamespaceLocked(ns.Name, false, ns)
```

Comment:
this will end up calling configureNamespace, which will potentially handle 3 things:
1. configuring exgw for all pods in this namespace
2. configuring network policy ACL logging for all pods in this namespace
3. configuring multicast for all pods in this namespace

nsInfo and network policy locking is heavily done here to make sure the pod and network policy threads to not step on the namespace thread while doing these operations. I think the question is do we really want to keep that approach? Or as @jcaamano suggested, we could just add the pods back to the pod workqueue and then the pod handler will eventually process them and potentially add the missing exgw routes/functionality?